polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -1680,6 +1682,61 @@ class RaiseSetCaptured : public OpHostRaisePattern<LLVM::VarAnnotation> {
     return {op, value};
   }
 
+  /// Return the `llvm.alloca` operation defining \p lambdaObj.
+  static LLVM::AllocaOp getAlloca(Value lambdaObj) {
+    if (Operation *definingOp = lambdaObj.getDefiningOp()) {
+      // lambdaObj defined in this function
+      return TypeSwitch<Operation *, LLVM::AllocaOp>(definingOp)
+          .Case<LLVM::AllocaOp>([](auto alloca) { // Return the defining alloca
+            return alloca.getElemType().has_value() ? alloca : nullptr;
+          })
+          // Safely bail out
+          .Default(LLVM::AllocaOp());
```

Comment:
```suggestion
          .Default(nullptr);
```
Nit: elsewhere you use `nullptr` to signal failure.

Reply:
Slightly changed so that it compiles